sympy/ntheory/generate.py

```diff
@@ -244,11 +247,35 @@ def primepi(n):
         prime : Return the nth prime
     """
     n = int(n)
-    if n < 2:
-        return 0
-    else:
-        return sieve.search(n)[0]
-
+    if n<2:
+		return 0	
+	lim = int(n**0.5) #lim = floor(sqrt(n))
```

Comment:
Also I can't tell for sure but it looks like having `lim` be exact might not be that important, as long as it's not too small.